### PR TITLE
Sql Migrations

### PR DESCRIPTION
**Pull request by RonAzar**

### CODE REVIEW

File: src/main/resources/db/migration/V202408091410__owner_table.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+create table owner(
+    owner_id serial primary key ,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091410__owner_table.sql:2]: rename to id, the owner part is redundant in the context of owner

File: src/main/resources/db/migration/V202408091410__owner_table.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    name varchar(100) not null ,
+    company_id int,
+    employee_id int
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091410__owner_table.sql:5]: employee_id is string in bob
also should be not null

File: src/main/resources/db/migration/V202408091410__owner_table.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+create table owner(
+    owner_id serial primary key ,
+    name varchar(100) not null ,
+    company_id int,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091410__owner_table.sql:4]: should be not null

File: src/main/resources/db/migration/V202408091410__owner_table.sql [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    employee_id int
+);
+
+CREATE INDEX idx_owner_id ON owner(owner_id);
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091410__owner_table.sql:8]: should be one index on both columns, and the name should include the table name

File: src/main/resources/db/migration/V202408091410__owner_table.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
+create table owner(
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091410__owner_table.sql:1]: the name of the file should be the date it was created

File: src/main/resources/db/migration/V202408271754__pets_table.sql [hunk 6]
```diff
@@ -0,0 +1,35 @@
+create table pets
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408271754__pets_table.sql:1]: same here

File: src/main/resources/db/migration/V202408271754__pets_table.sql [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+(
+    id serial primary key ,
+    name varchar(100) not null ,
+    type varchar(100) not null ,
+    company_id int,
+    date_of_arrival date not null
+);
+
+CREATE INDEX idx_company_id ON pets(company_id);
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408271754__pets_table.sql:10]: index should include the table name

File: src/main/resources/db/migration/V202408271754__pets_table.sql [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+--
+-- --Retrieve the pets by type (dog, cat..)
+--
+-- Select Distinct type
+-- From pets
+--
+--
+-- --Delete pet by id
+--          Delete from pets
+-- Where (id == 2)
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408271754__pets_table.sql:29]: where id=2

**Review by mayhuberman — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V202408091410__owner_table.sql (hunk 5 above)
Comment (RonAzar) [reply src/main/resources/db/migration/V202408091410__owner_table.sql:1]: Ok,
Got it!

File: src/main/resources/db/migration/V202408091410__owner_table.sql (hunk 1 above)
Comment (RonAzar) [reply src/main/resources/db/migration/V202408091410__owner_table.sql:2]: got it!

File: src/main/resources/db/migration/V202408091410__owner_table.sql [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+    id serial primary key ,
+    name varchar(100) not null ,
+    company_id int not null,
+    employee_id varchar(100) not null
+);
+
+CREATE INDEX idx_owner_id_employee_id ON owner(id, employee_id);
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091410__owner_table.sql:8]: There is no point in putting an index on a primary key
it should be on company_id and employee_id

File: src/main/resources/db/migration/V202408091410__owner_table.sql [hunk 10]
```diff
@@ -0,0 +1,8 @@
+create table owner(
+    id serial primary key ,
+    name varchar(100) not null ,
+    company_id int not null,
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/resources/db/migration/V202408091410__owner_table.sql:4]: fixed it

File: src/main/resources/db/migration/V202408091410__owner_table.sql (hunk 9 above)
Comment (RonAzar) [reply src/main/resources/db/migration/V202408091410__owner_table.sql:8]: Oops...
Got it!

File: src/main/resources/db/migration/V202408271754__test.sql [hunk 11]
```diff
@@ -1,5 +1,4 @@
-create table test_me
-(
+create table test_me(
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/resources/db/migration/V202408271754__test.sql:1]: I’m not sure I fully understand.
Could you clarify what I should do?

[the remark this answers is not in the document]
Comment (mayhuberman) [reply src/main/resources/db/migration/V202408271754__test.sql:1]: do not touch this file 😅

[the remark this answers is not in the document]
Comment (mayhuberman) [reply src/main/resources/db/migration/V202408271754__test.sql:1]: there are changes on this file

Comment (mayhuberman) [inline src/main/resources/db/migration/V202408271754__test.sql:1]: remove all changes from this migration
it should be like this
```
create table test_me
(
```

Comment (RonAzar) [reply src/main/resources/db/migration/V202408271754__test.sql:1]: Changed

**Review by mayhuberman — APPROVED** [review]

**Pull request merged**